From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Wed, 10 Jan 2024 20:20:56 +0100
Subject: [PATCH] Modifiable Bukkit enums - Inject runtime versions - Define
 Fiddle enum name prefix

Defines the prefix that will be used in the names of enum members added by Fiddle.

Also adds a reference to it in the `Material.matchMaterial` documentation, as it is a relevant example of the possible matching.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/org/bukkit/Material.java b/src/main/java/org/bukkit/Material.java
index 2188f12407bb299bcef22c84e33cb27149066cdd..d9d47a1d63101ab701960e109ca38f6b28dbe6fa 100644
--- a/src/main/java/org/bukkit/Material.java
+++ b/src/main/java/org/bukkit/Material.java
@@ -5055,7 +5055,7 @@ public enum Material implements org.fiddlemc.fiddle.material.AbstractMaterial {
      * <ol>
      *     <li>The {@linkplain #getKey() namespaced key}, for example: <code>"minecraft:stone_bricks"</code> or <code>"masonry:tuff_bricks"</code></li>
      *     <li>The namespaced key path (part behind the : in a namespaced key), for example: <code>"stone_bricks"</code> or <code>"tuff_bricks"</code></li>
-     *     <li>The enum name, for example: <code>"STONE_BRICKS"</code></li>
+     *     <li>The enum name, for example: <code>"STONE_BRICKS"</code> or <code>"{@linkplain org.fiddlemc.fiddle.material.FiddleEnumNamePrefix#PREFIX FIDDLE_}MASONRY_TUFF_BRICKS"</code></li><!-- // Fiddle - modifiable Bukkit enums - inject runtime versions - define Fiddle enum name prefix -->
      * </ol>
      * In other words, first the given string will be checked case-sensitively against the namespaced key,
      * then case-sensitively against the namespaced key path, and so on.
diff --git a/src/main/java/org/fiddlemc/fiddle/material/FiddleEnumNamePrefix.java b/src/main/java/org/fiddlemc/fiddle/material/FiddleEnumNamePrefix.java
new file mode 100644
index 0000000000000000000000000000000000000000..c2c19b7f657061878b5ecd66d3bc185c6d74fb23
--- /dev/null
+++ b/src/main/java/org/fiddlemc/fiddle/material/FiddleEnumNamePrefix.java
@@ -0,0 +1,16 @@
+// Fiddle - modifiable Bukkit enums - inject runtime versions - define Fiddle enum name prefix
+
+package org.fiddlemc.fiddle.material;
+
+/**
+ * A utility class solely for holding the {@link #PREFIX} value.
+ */
+public final class FiddleEnumNamePrefix {
+
+    private FiddleEnumNamePrefix() {
+        throw new RuntimeException();
+    }
+
+    public static final String PREFIX = "FIDDLE_";
+
+}
